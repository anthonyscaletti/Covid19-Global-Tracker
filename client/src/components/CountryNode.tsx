import React from 'react';

import '../styles/countryNode.css';

const CountryNode = (props: IOwnProps) => {
    const isSelectedStyle = (props.selectedCountryCode === props.countryCode) ? {backgroundColor: "#8AAE5C"} : {};

    return (
        <div className="countryNode" 
             onClick={() => props.onClick(props.countryCode)}
             style={isSelectedStyle}
        >
            <h5>&#x25C9;&nbsp;&nbsp;{props.country}:<span style={{float: "right", marginRight: "5%"}}>{props.value}</span></h5>
        </div>
    );
}

interface IOwnProps {
    country: string,
    countryCode: string,
    selectedCountryCode: string,
    value: number,
    onClick: Function
}
    
export default CountryNode;