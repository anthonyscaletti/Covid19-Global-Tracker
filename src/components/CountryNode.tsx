import React from 'react';

import '../styles/countryNode.css';

export interface OwnProps {
    country: string,
    countryCode: string,
    totalConfirmed: number
}

const CountryNode = (props: OwnProps) => {
    return (
        <div className="countryNode">
            <h5>&#x25C9;&nbsp;&nbsp;{props.country}:<span style={{float: "right", marginRight: "5%"}}>{props.totalConfirmed}</span></h5>
        </div>
    );
}
    
export default CountryNode;