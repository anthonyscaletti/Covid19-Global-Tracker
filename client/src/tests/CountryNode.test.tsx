import React from 'react';
import renderer from 'react-test-renderer';

import CountryNode from '../components/CountryNode';

it('renders correctly', () => {
  const tree = renderer
    .create(<CountryNode />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});