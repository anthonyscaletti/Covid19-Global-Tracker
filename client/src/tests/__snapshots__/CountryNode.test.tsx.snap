// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="countryNode"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#8AAE5C",
    }
  }
>
  <h5>
    ◉  
    :
    <span
      style={
        Object {
          "float": "right",
          "marginRight": "5%",
        }
      }
    />
  </h5>
</div>
`;
